super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,31 +231,65 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
-				if( c == quoteChar ) {
+
+				if( c == quoteEscapeChar ) {
 					int nextCharIndex = charIndex + 1;
 					boolean availableCharacters = nextCharIndex < line.length();
 					boolean nextCharIsQuote = availableCharacters && line.charAt(nextCharIndex) == quoteChar;
+					boolean nextCharIsEscapeQuoteChar = availableCharacters && line.charAt(nextCharIndex) == quoteEscapeChar;
+
 					if( nextCharIsQuote ) {
 						/*
-						 * An escaped quote (""). Add a single quote, then move the cursor so the next iteration of the
-						 * loop will read the character following the escaped quote.
+						 * An escaped quote (e.g. "" or \"). Skip over the escape char, and add
+						 * the following quote char as part of the column;
+						 */
+						charIndex++;
+						currentColumn.append(line.charAt(nextCharIndex));
+					} else if( nextCharIsEscapeQuoteChar ) {
+						/*
+						 * A double escape (normally \\). Save the escape char, then continue to
+						 * next character.
 						 */
 						currentColumn.append(c);
 						charIndex++;
-						
-					} else {
+					} else if( quoteEscapeChar == quoteChar ) {
```

Comment:
I dont understand what this branch is doing here. It is irregardles of the current read char or the next char. If the `quoteEscapeChar == quoteChar` then how can we even get to this branch. It is an `else ` of `if( nextCharIsQuote ) {` ...